src/repl/mod.rs

```diff
@@ -230,7 +232,9 @@ impl Repl for ReplImpl {
         for id in &pending {
             self.cache.resolve_imports(*id).unwrap();
         }
-        typecheck::type_check_in_env(&term, &self.env.type_env, &self.cache)?;
+        // Substitute the wildcard types for their inferred types
+        let (_, wildcards) = typecheck::type_check_in_env(&term, &self.env.type_env, &self.cache)?;
+        let term = transform::substitute_wildcards::transform_one(term, &wildcards);
```

Comment:
Actually, no, good point.  I looked into this, and indeed I think my assumption is wrong, because if the term is a `MetaValue` with a `Flat` type or with an inner value that isn't `None`, then we'd need to recurse into it to get the proper type.